src/config.rs

```diff
@@ -0,0 +1,313 @@
+
+use std::fs::File;
+use std::io::{Read, Error};
+use std::path::{PathBuf, Path};
+use std::default::Default;
+use std::fmt;
+
+use yaml_rust::{Yaml, YamlLoader, ScanError};
+
+pub enum ConfigError {
+    YamlScan(ScanError),
+    YamlParse(ConfigMeta),
+    FileError(String, Error),
+    MissingExternalFile(ConfigMeta),
+}
+
+impl ConfigError {
+    pub fn to_string(&self) -> String {
+        match self {
+            &ConfigError::YamlScan(ref e) => format!("Failed to scan YAML: {}", e),
+            &ConfigError::YamlParse(ref meta) => {
+                let mut path = String::new();
+
+                for (index, element) in meta.fields.iter().enumerate() {
+                    if index != 0 {
+                        path = path + "->";
+                    }
+
+                    path = path + element;
+                }
+
+                format!("{}: Failed to parse YAML: {}: expect {}", meta.path.display(), path, meta.ty)

```

Comment:
Should probably re-add in a check if the key is a "BadValue", if it then say "Could not find YAML" instead of "Failed to parse YAML".